strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BaseController.java

```diff
@@ -297,8 +299,20 @@ protected void copyToResponse(InputStream is,
             while ((readLength = is.read(bytes, 0, bytes.length)) != -1)
             {
                 // Write the artifact
-                os.write(bytes, 0, readLength);
-                os.flush();
+                try
+                {
+                    os.write(bytes, 0, readLength);
+                    os.flush();
+                }
+                catch (EOFException e)
+                {                   
+                    if (e.getCause().getMessage().equals("Connection reset by peer")) {
```

Comment:
please also check if `IOException.class.equals(e.getCause().getClass())`, There are other possibilities that may cause `Connection reset by peer` message, see this thread for example: https://stackoverflow.com/questions/20741405/javax-net-ssl-sslexception-ssl-handshake-aborted-connection-reset-by-peer-while/20745624

Reply:
Thanks, man! I really appreciate your help! My intention here was to catch  explicitly EOFException, which extends IOException. I was sure that all other exceptions can't be catches with this code. And only after that i check message. Do i really need to check class?

Reply:
Sorry, but I meant `e.getCause()`, not `e`.